pipeline-yaml_5b5744d40428631d7a8940a9.md

```diff
@@ -0,0 +1,305 @@
+- [About Pipeline YAML](#pipeline-yaml-reference)
+- [Version 1](#version-1)
+  * [Index](#index)
+  * [The preamble](#the-preamble)
+  * [The agent](#the-agent-property)
+  * [Preamble example](#preamble-example)
+
+
+- [Version 2](#version-2)
+  * [Constants](#constants)
+  * [See also](#see-also)
+
+- [Version 3](#version-3)
+  * [Properties](#properties)
+  * [The fields of the preamble](#the-fields-of-the-preamble)
+  * [The fields of the agent property](#the-fields-of-the-agent-property)
+
+
+
+# Pipeline YAML Reference
+
+This document is the reference for the YAML grammar used for describing the pipelines of Semaphore 2.0.
+
+
+# Version 1
+
+
+## Index
+
+* [agent](#the-agent-property)
+* [blocks](#the-blocks-property)
+* [os_image](#the-os_image-property)
+* [type](#the-type-property)
+* [version](#the-version-property)
+
+## Constants
+
+* [e1-standard-2](#the-e1-standard-2-value)
+* [ubuntu1804](#the-ubuntu1804-value)
+
+## Examples
+* [Preamble example](#preamble-example)
+
+
+## The preamble
+
+Each Semaphore 2.0 pipeline configuration file has a preamble, which is
+mandatory and should be present on every `.semaphore/semaphore.yml` file you create.
+
+
+## The version property
+
+The `version` property is a string value that shows the version of the
+pipeline YAML file that will be used.
+
+List of valid values: `"v1.0"`
+
+## The name property
+
+The `name` property is a Unicode string that assigns a name to the pipeline.
+
+*Caution*: The `name` property can be found in other sections because it can also
+be used for defining the name of a job inside a `jobs` block or the name of
+a `task` section defined using `task`.
+
+
+
+## The agent property
+
+The `agent` property requires three additional properties that identify the
+environment (type of container) in which the jobs of the pipeline will get executed.
```

Comment:
@mactsouk Lets stay away from using term `container`. It is something that we use from time to time internally but  for someone outside it might be confusing because they could connect it with "Docker container".

Maybe switch `(type of container)` with more precise "definition" of virtual machine hardware + virtual machine operating system.